lib/avro/builder/field.rb

```diff
@@ -84,14 +86,18 @@ def name(value = nil)
       end
 
       def serialize(reference_state)
-        # TODO: order is not included here
-        {
+        attrs = {
           name: name,
           type: serialized_type(reference_state),
-          doc: doc,
-          default: default,
           aliases: aliases
-        }.reject { |_, v| v.nil? }.tap do |result|
+        }
+
+        # TODO: order is not included here
+        self.class.dsl_attribute_names.reject { |attr| [:order, :type].include?(attr) }.each do |attr|
+          attrs[attr] = send(attr)
+        end
+
+        attrs.reject { |_, v| v.nil? }.tap do |result|
```

Comment:
The fields support either options/args on the `field` call or can be defined with a block.

To support the block I think you'll need to also update the code further up this file in order to define the appropriate methods. Around line 34.

Reply:
e.g:
```
field :blah do
  doc "Soemthing"
  myNewField 12
end

Reply:
Discussed in Slack, it looks to work:

```ruby
required :locale_code, :string do
  doc "An ISO639 locale code of the translation."
  sensitivityLevel "none"
end
```